explorer/nodewatch/tests/test_app.py

```diff
@@ -184,12 +184,106 @@ def test_get_api_symbol_nodes_peer(client):  # pylint: disable=redefined-outer-n
 	# Assert: spot check names
 	assert 200 == response.status_code
 	assert 'application/json' == response.headers['Content-Type']
-	assert 5 == len(response_json)
+	assert 7 == len(response_json)
 	assert [
-		'Apple', 'Shin-Kuma-Node', 'ibone74', 'jaguar', 'symbol.ooo maxUnlockedAccounts:100'
+		'Apple', 'Shin-Kuma-Node', 'ibone74', 'jaguar', 'symbol.ooo maxUnlockedAccounts:100', 'xym pool', 'yasmine farm'
 	] == list(map(lambda descriptor: descriptor['name'], response_json))
 
 
+def test_get_api_symbol_nodes(client):  # pylint: disable=redefined-outer-name
+	# Act:
+	response = client.get('/api/symbol/nodes')
+	response_json = json.loads(response.data)
+
+	# Assert: spot check names
+	assert 200 == response.status_code
+	assert 'application/json' == response.headers['Content-Type']
+	assert 8 == len(response_json)
+	assert [
+		'Allnodes250', 'Apple', 'Shin-Kuma-Node', 'ibone74', 'jaguar', 'symbol.ooo maxUnlockedAccounts:100', 'xym pool', 'yasmine farm'
+	] == list(map(lambda descriptor: descriptor['name'], response_json))
+
+
+def test_get_api_symbol_nodes_order_random(client):  # pylint: disable=redefined-outer-name
+	# Act:
+	response = client.get('/api/symbol/nodes?order=random')
+	response_json = json.loads(response.data)
+
+	# Assert: spot check names
+	assert 200 == response.status_code
+	assert 'application/json' == response.headers['Content-Type']
+	assert 8 == len(response_json)
+	expected_names = [
+		'Allnodes250', 'Apple', 'Shin-Kuma-Node', 'ibone74', 'jaguar', 'symbol.ooo maxUnlockedAccounts:100', 'xym pool', 'yasmine farm'
+	]
+	actual_names = list(map(lambda descriptor: descriptor['name'], response_json))
+	assert sorted(expected_names) == sorted(actual_names)
+
+
+def test_get_api_symbol_nodes_limit_2(client):  # pylint: disable=redefined-outer-name
+	# Act:
+	response = client.get('/api/symbol/nodes?limit=2')
+	response_json = json.loads(response.data)
+
+	# Assert: spot check names
+	assert 200 == response.status_code
+	assert 'application/json' == response.headers['Content-Type']
+	assert 2 == len(response_json)
+	assert [
+		'Allnodes250', 'Apple'
+	] == list(map(lambda descriptor: descriptor['name'], response_json))
+
+
+def test_get_api_symbol_nodes_ssl_true(client):  # pylint: disable=redefined-outer-name
+	# Act:
+	response = client.get('/api/symbol/nodes?ssl=true')
+	response_json = json.loads(response.data)
+
+	# Assert: spot check names
+	assert 200 == response.status_code
+	assert 'application/json' == response.headers['Content-Type']
+	assert 1 == len(response_json)
+	assert [
+		'xym pool'
+	] == list(map(lambda descriptor: descriptor['name'], response_json))
+
+
+def test_get_api_symbol_nodes_ssl_false(client):  # pylint: disable=redefined-outer-name
```

Comment:
why would we ever want a route like this? i don't see a compelling need for filtering nodes without SSL.

also, imo, it should be inverse of true (i.e. contain all nodes without apiNodeInfo.

Reply:
yeah, you are right, `ssl=false` is not needed actually. 
should we make it as default filter all `ssl=true`?

Reply:
no - i think default should return all
and there is optional filter to return only ssl compatible nodes